kanidm_unix_int/src/daemon.rs

```diff
@@ -374,9 +370,63 @@ async fn main() {
     let cgid = get_current_gid();
     let cegid = get_effective_gid();
 
-    if cuid == 0 || ceuid == 0 || cgid == 0 || cegid == 0 {
-        eprintln!("Refusing to run - this process must not operate as root.");
-        std::process::exit(1);
+    let clap_args = Command::new("kanidm_unixd")
+        .version(env!("CARGO_PKG_VERSION"))
+        .about("Kanidm Unix daemon")
+        .arg(
+            Arg::new("skip-root-check")
+                .help("Allow running as root")
+                .short('r')
+                .long("skip-root-check")
+                .action(ArgAction::SetTrue),
+        )
+        .arg(
+            Arg::new("debug")
+                .help("Show extra debug information")
+                .short('d')
+                .long("debug")
+                .action(ArgAction::SetTrue),
+        )
+        .arg(
+            Arg::new("configtest")
+                .help("Display the configuration and exit")
+                .short('t')
+                .long("configtest")
+                .action(ArgAction::SetTrue),
+        )
+        .arg(
+            Arg::new("unixd-config")
+                .takes_value(true)
+                .help("Set the unixd config file path")
+                .short('u')
+                .long("unixd-config")
+                .default_value(DEFAULT_CONFIG_PATH)
+                .env("KANIDM_UNIX_CONFIG")
+                .action(ArgAction::StoreValue),
+        )
+        .arg(
+            Arg::new("client-config")
+                .takes_value(true)
+                .help("Set the client config file path")
+                .short('c')
+                .long("client-config")
+                .default_value(DEFAULT_CLIENT_CONFIG_PATH)
+                .env("KANIDM_CLIENT_CONFIG")
+                .action(ArgAction::StoreValue),
+        )
+        .get_matches();
+
+    if clap_args.get_flag("skip-root-check") {
+        warn!("Skipping root user check, if you're running this for testing, ensure you clean up temporary files.")
+        // TODO: this wording is not great m'kay.
+    } else {
+        if cuid == 0 || ceuid == 0 || cgid == 0 || cegid == 0 {
+            error!("Refusing to run - this process must not operate as root.");
+            return;
+        }
+    };
+    if clap_args.get_flag("debug") {
+        std::env::set_var("RUST_LOG", "DEBUG");
```

Comment:
minor nit, debug should be lower case.

Reply:
Do you mean in the set environment variable? It's lowercase in the highlighted line...?

Reply:
Yeah, I meant in ('RUST_LOG', 'debug')

I still have NFI how to "add suggestions" that you can commit else I would have.

Reply:
Click the line to highlight, then this widget in the comment box 😄 : ![Screen Shot 2022-09-08 at 10 29 16](https://user-images.githubusercontent.com/168188/189007653-56bab2e5-8518-4d1c-b33c-2d2c1a76284a.png)

Sorted!